src/hardware/serialport/softmodem.cpp

```diff
@@ -200,8 +271,8 @@ bool CSerialModem::Dial(char * host) {
 		port = MODEM_DEFAULT_PORT;
 
 	// Resolve host we're gonna dial
-	LOG_MSG("Connecting to host %s port %u", host, port);
-	clientsocket.reset(new TCPClientSocket(host, port));
+	LOG_MSG("Connecting to host %s port %u", buf, port);
+	clientsocket.reset(new TCPClientSocket(buf, port));
 	if (!clientsocket->isopen) {
```

Comment:
The `host` variable is more instructive than `buf`; after all - we're telling the TCP Socket to "use this hostname", as opposed to "here's a buffer for you to do stuff with".  Can you re-work the names a bit here?